onnxruntime/core/session/provider_bridge_ort.cc

```diff
@@ -213,6 +253,122 @@ common::Status LoadDynamicLibraryFromProvider(onnxruntime::PathString library_na
 struct ProviderHostImpl : ProviderHost {
   const OrtApiBase* OrtGetApiBase() override { return ::OrtGetApiBase(); }
 
+  Status GetEPOptimizerByName(const std::string& name,
+                              const GraphTransformerManager& transformer_mgr,
+                              std::function<std::vector<std::unique_ptr<ComputeCapability>>(const GraphViewer&)>& selection_func) override {
+    static const GraphTransformerManager& graph_transformer_mgr = transformer_mgr;
+    std::string optimizer_name(name);
+
+    // Pre-defined graph transformers/optimizers
+    static const std::string kEP_GRAPH_TRANSFORMER_CONSTANT_FOLDING_DQ = "ConstantFoldingDQ";
+
+    // ConstantFoldingDQ optimization function
+    auto constant_folding_dq_optimization = [&](Graph& graph, const ComputeCapability& optimization_cc, ComputeCapability& cc_to_update) -> Status {
+      std::string optimizer_name = kEP_GRAPH_TRANSFORMER_CONSTANT_FOLDING_DQ;
+      auto logger = const_cast<logging::Logger*>(&logging::LoggingManager::DefaultLogger());
+      std::unordered_set<std::string> original_initializers_to_remove;
+      std::unordered_set<std::string> new_initializers_to_add;
+      InlinedHashSet<NodeIndex> dq_node_index_set;
+
+      // iterate node_to_optimize to:
+      //   1. get original initializers to remove 
+      //   2. add new initializers
+      //   3. create dq node index set
+      for (const auto& index : optimization_cc.sub_graph->nodes) {
+        auto node = graph.GetNode(index);
+        if (node->OpType() != "DequantizeLinear") {
+          continue;
+        }
+        auto input_0 = node->InputDefs()[0];
+        auto output_0 = node->OutputDefs()[0];
+        original_initializers_to_remove.insert(input_0->Name());
+        new_initializers_to_add.insert(output_0->Name());
+        dq_node_index_set.insert(index);
+      }
+
+      ConstantFoldingDQ* transformer = static_cast<ConstantFoldingDQ*>(graph_transformer_mgr.GetTransformerByName(optimizer_name));
+      transformer->UpdateNodeIndexSet(dq_node_index_set);
+      
+      // apply constant folding on DQ nodes
+      graph_transformer_mgr.ApplyTransformer(graph, optimizer_name, *logger);
+
+      // update the overall ComputeCapability
+      std::vector<onnxruntime::NodeIndex> updated_nodes;
+      for (auto index : cc_to_update.sub_graph->nodes) {
+        if (dq_node_index_set.find(index) != dq_node_index_set.end()) {
+          continue;
+        }
+        updated_nodes.push_back(index);
+      }
+      cc_to_update.sub_graph->nodes = updated_nodes;
+
+      auto original_meta_def = cc_to_update.sub_graph->GetMetaDef();
```

Comment:
Instead of creating a new MetaDef, now that we have a use-case where it makes sense to update it could we instead add a method to get a mutable MetaDef from sub_graph?

Reply:
added `GetMutableMetaDef`